### PR TITLE
Adds (missing) base file to list of all parsed files

### PR DESCRIPTION
**Pull request by dleffler**
($compiler->parsedFiles). essential for successful cache operation. #339

### CODE REVIEW

**Comment by mahagr** [comment]
Is there any change to put this addition into its own function. I'm overriding file cache with my own stream implementation, which changes the filenames into URL format -- thus using `realpath()` would make those to fail.

**Comment by dleffler** [comment]
Yes, that's how it's implemented in lessphp.  I'll work on that.

**Comment by dleffler** [comment]
Fix for issues #339 and #337

**Comment by robocoder** [comment]
Can you split this into 2 separate PRs?

Your fix for #337 9863a8f breaks a different scope test (https://travis-ci.org/leafo/scssphp/jobs/81457392).

File: src/Compiler.php [hunk 1]
```diff
@@ -2502,10 +2504,10 @@ public function get($name, $shouldThrow = true, $env = null)
...
             }
 
-            if (! $hasNamespace && isset($env->marker)) {
-                $env = $this->rootEnv;
-                continue;
-            }
+//            if (! $hasNamespace && isset($env->marker)) {
+//                $env = $this->rootEnv;
+//                continue;
+//            }
```
Comment (robocoder) [inline src/Compiler.php]: This commit 9863a8f3b4f883f54e393e63228f0b3545f5152c breaks one or more variable scope tests.

File: src/Compiler.php [hunk 2]
```diff
@@ -2585,6 +2587,19 @@ public function unsetVariable($name)
     }
 
     /**
+     * Adds to list of parsed files
+     *
+     * @api
+     *
+     * @param string $file
+     */
+    public function addParsedFile($file) {
```
Comment (robocoder) [inline src/Compiler.php]: If you run `vendor/bin/phpcs --standard=PSR2`, you'll see one or more coding standard violations.

File: src/Compiler.php [hunk 3]
```diff
@@ -2585,6 +2587,19 @@ public function unsetVariable($name)
...
+        if (!empty($file)) {
```
Comment (robocoder) [inline src/Compiler.php]: Won't work ... `bin/pscss` passes `'STDIN'` when there's no file.

**Comment by dleffler** [comment]
I hate pull requests :)

**Comment by robocoder** [comment]
No worries if you can't get to it right away. I can cherry-pick and make the necessary changes.

**Pull request closed without merging**